pkg/workspace/template_loader.go

```diff
@@ -160,7 +160,7 @@ func (l *TemplateLoader) EvalYAML(libraryCtx LibraryExecutionContext, file *file
 	l.ui.Debugf("### ast\n")
 	docSet.Print(l.ui.DebugWriter())
 
-	if !file.IsTemplate() && !file.IsLibrary() {
+	if !file.IsTemplate() && !file.IsLibrary() || !yamltemplate.HasTemplating(docSet) {
```

Comment:
i was thinking this reads better:

```
if !file.IsTemplate() && !file.IsLibrary() {
    return nil, docSet, nil
}
if !yamltemplate.HasTemplating(docSet) {
    return nil, docSet, nil
}
```

Reply:
I respectfully disagree.

The way it's written in the PR, it's the same early exit for the same reason and it's all in one sentence:

"If the file is not a template... and it's not a library... Ooooor, it just doesn't have any templating in the first place... well, no need to 'evaluate' it."

Reply:
ok lets keep it as is.